lib/charms/grafana_agent_k8s/v0/grafana_agent.py

```diff
@@ -127,9 +141,10 @@ def __init__(self, charm, log_files: list, relation_name: str = DEFAULT_RELATION
         self._stored.set_default(grafana_agents="{}")
         self._charm = charm
         self._relation_name = relation_name
+        containers = dict(self._charm.model.unit.containers)
+        self._container_name = [*containers].pop()
```

Comment:
This dangerously assumes that the container we care about is the first. This is not a healthy assumption unless there is exactly ONE container. Please add the container name as an optional argument, and is `None` is provided, check if there is just ONE container and if not fail hard.

Reply:
Sure, I harcoded this to the first container because I wanted something that just work to continue with the development.